src/components/src/side-panel/common/types.ts

```diff
@@ -1,8 +1,21 @@
-import React, {MouseEvent} from 'react';
+import React from 'react';
 import {openDeleteModal, VisStateActions, ActionHandler} from '@kepler.gl/actions';
 import {RGBColor} from '@kepler.gl/types';
 import KeplerTable, {Datasets} from '@kepler.gl/table';
 
+type MiniDataset = {
+  id: string;
+  color: RGBColor;
+  label?: string;
+};
+
+export type PanelMeta = {
+  id: string;
+  label: string;
+  iconComponent: React.ElementType;
+  onClick: null;
+};
+
 export type DatasetInfoProps = {
```

Comment:
this should be 
`dataset: MiniDataset`?

Reply:
As I remember it caused ts issues with related components, as something was using fields from KeplerDataset, not available in MiniDataset. I'll check again.

Reply:
If that's case, we should add `fields: MiniField[]` to `MiniDataset`